### PR TITLE
Join tables 

### PR DESCRIPTION
**Pull request by SabaAhmad404**
### In this branch, I have worked on the following things :+1: 

:heavy_check_mark: Added Many to Many Relationships
:heavy_check_mark: Create Specializations, Visits Tables
:heavy_check_mark: Create Primary & Foreign Keys
:heavy_check_mark: Added Data to Specializations, Visits Tables
:heavy_check_mark: Added SQL Queries to Get Analytic Data

**1st query**
image

**2nd query**
image

**3rd query**
image

**4th query**

image

**5th query**

image

**6th query**

image

**7th query**

image

**8th query**
image

**9th query**

image

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -1,3 +1,244 @@
...
+inner join vets on visits.vets_id=vets.id
+where vets.name='Stephanie Mendez'
+and visits.date_of_vists BETWEEN '01-04-2020' AND '08-30-2020'
+
+-- What animal has the most visits to vets?
+Select count(animals.name), vets.name from animals inner join visits on animals.id=visits.animal_id
+inner join vets on vets.id= visits.vets_id
+group by vets.name
+order by count(animals.name) desc
+limit 1
```
Comment (mikethreels) [inline queries.sql]: - [ ] in this query you are supposed to return the name of an animal whit the most visits to a vet, now it returns a count of animals who visited the vet with the most visits

File: queries.sql [hunk 2]
```diff
@@ -1,3 +1,244 @@
...
+
+-- How many visits were with a vet that did not specialize in that animal's species?
+
+SELECT COUNT(visits.date_of_vists)-(SELECT COUNT(vets.name) 
+FROM vets, specializations, animals, visits
+WHERE
+visits.vets_id = vets.id
+AND animals.id = visits.animal_id
+AND concat(animals.species_id, visits.vets_id) = concat(specializations.species_id, specializations.vets_id)) total_unspecialized_cases 
+FROM visits;
```
Comment (mikethreels) [inline queries.sql:233]: - [ ] only Maisy Smith doesn't specialize in a species and she has 8 visits to her name, and the result of this query is 13.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Saba, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 2 above)
Comment (SabaAhmad404) [reply queries.sql:233]: @mikethreels 

this is the query

  SELECT animals.name, visits.date_of_vists, vets.name, visits.vets_id, animals.species_id
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vets_id
  INNER JOIN species ON species.id = animals.species_id


image
image

I hope you are doing well. I have taken the screenshot and highlighted those visits which were non-specialized vets visits. Kindly count them. It is not just about Maisy smith Actually other vets who are not specialized in other species they have done visits to their animals too.

**Review by DeliceLydia — APPROVED** [review]
## Hi @SabaAhmad404, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- Addressed changes requested

### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**